### ISSUE TITLE
[🐛 Bug]: Coverage does not work when using `browser.execute(Async)`

### ISSUE THREAD

**Issue by Mrtenz** [webdriverio/webdriverio#9849]
### Have you read the Contributing Guidelines on issues?

- [X] I have read the [Contributing Guidelines on issues](https://github.com/webdriverio/webdriverio/blob/main/CONTRIBUTING.md#reporting-new-issues).

### WebdriverIO Version

8.3.10

### Node.js Version

16.19.1

### Mode

WDIO Testrunner

### Which capabilities are you using?

```typescript
{
  browserName: 'chrome',
},
{
  browserName: 'firefox',
},
```

### What happened?

When using the `browser` runner with the coverage option enabled, `browser.execute(Async)` does not work anymore. Instead, it will throw an error:

```
Failed to execute command "executeAsyncScript": ReferenceError: cov_fdlvqmfo is not defined
```

This causes the entire test to fail.

### What is your expected behavior?

The test should work, regardless of whether `execute(Async)` is used.

### How to reproduce the bug.

1. Use the `browser` runner with coverage enabled:
   ```ts
   export const config: Options.Testrunner = {
     /* ... */
     runner: [
       'browser',
       {
         coverage: {
           enabled: true,
         },
       },
     ],
     /* ... */
   }
   ```
2. Create a test file, and add the following test:
   ```ts
   it('logs "Hello world" to the console', async () => {
     await browser.executeAsync(() => {
       console.log('Hello world');
     });
   });
   ```
3. Run Webdriverio with `wdio run [config]`.

### Relevant log output

```typescript
Error: Failed to execute command "executeAsyncScript": javascript error: cov_fdlvqmfo is not defined
JavaScript stack:
ReferenceError: cov_fdlvqmfo is not defined
    at eval (eval at executeAsyncScript (:552:26), <anonymous>:4:5)
    at eval (eval at executeAsyncScript (:552:26), <anonymous>:7:6)
    at eval (eval at executeAsyncScript (:552:26), <anonymous>:7:31)
    at executeAsyncScript (<anonymous>:552:47)
    at <anonymous>:567:29
    at callFunction (<anonymous>:446:22)
    at <anonymous>:460:23
    at <anonymous>:461:3
  (Session info: headless chrome=110.0.5481.177)
    ✖ logs "Hello world" to the console » [  ]
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Is there an existing issue for this?

- [X] I have searched the existing issues

**Comment by christian-bromann** [comment]
Given the code is taken and run within a different context you should do the following:

```ts
  await browser.executeAsync(/* istanbul ignore if */() => {
    console.log('Hello world');
  });
```

Let me see if we can automatically set these comments through Vite.

**Comment by Mrtenz** [comment]
I see, that makes a lot of sense! I was using the default `exclude` option, but it does not include `.test.browser.ts` (which is how I named my tests), so it was not excluding it from coverage. Adding `'**/*.test.browser.ts'` to the `exclude` array does indeed solve it.

I think some documentation about this could be useful, but other than that I don't think there's a need to change anything.

**Comment by Mrtenz** [comment]
I see you just opened a PR with some documentation. Thanks for handling this so quickly!

**Comment by christian-bromann** [comment]
> but it does not include `.test.browser.ts`

You can call your browser tests `*.browser.test.ts` which will automatically be excluded.

> I see you just opened a PR with some documentation.

Yes, added a page to the docs that will be merged once #9821 lands
